Update GDExtension tutorial with the hot reload feature (#9618)
godotengine/godot-docs: tutorials/scripting/gdextension/gdextension_cpp_example.rst

    [configuration]

    entry_symbol = "example_library_init"

>     compatibility_minimum = "4.2"

Reviewer: Is there a reason to bump this? I would rather have people extend their compatibility ranges than chasing latest. Maybe the trade-off could be elaborated in the text.

This file contains a ``configuration`` section that controls the entry function of the module.
You should also set the minimum compatible Godot version with ``compatability_minimum``,
which prevents older version of Godot from trying to load your extension.

> The ``reloadable`` flag enables your extension to be automatically loaded by the editor every time you recompile it,

Reviewer 2: ```suggestion
The ``reloadable`` flag enables your extension to be automatically reloaded by the editor every time you recompile it,
```

Author: Is there an easy way to squash these commits on github without needing to clone the repo and do it locally?

Reviewer 2: I don't think so to be honest. The only thing is to have GitHub do it with Merge and Squash but that's not allowed here due to the commit messages being just merged together which isn't clean. You will need to do that locally then

> The ``reloadable`` flag enables your extension to be automatically reloaded by the editor every time you recompile it,

Reviewer 3: ```suggestion
The ``reloadable`` flag enables automatic reloading of your extension by the editor every time you recompile it,
```

> without needing to restart the editor. This only works if you compile your extension in debug mode.

Reviewer: Maybe mention "loaded by the editor". I had lots of user being confused why "hot reloading" cannot reload in a running game. The term that Godot chose here is rather unfortunate.

when you run your project, you will see that our Godot icon travels along a
larger figure.

.. note::

    The hot reloading feature only works of you compiled your library with debug mode (default).
    Also make sure it's enabled in the .gdextension file as shown above.

> 

Reviewer 2: This seem to be doubled with the changes above. I would rather specify the section in the ``gdextension file`` explanation of the ``[configuration]`` section
